timezone/src/modules/Timezone.rb

```diff
@@ -984,6 +986,23 @@ def Summary
       HTML.List(ret)
     end
 
+
+    # Checks whether the system has Windows installed on a primary partition with NTFS
+    def system_has_windows?
+      storage = Y2Storage::StorageManager.instance
+      Storage::Ntfs::all(storage.probed).each do |ntfs|
+        ntfs.blk_devices.each do |blk_device|
+          if Storage::partition?(blk_device)
+            if Storage::to_partition(blk_device).type == Storage::PartitionType_PRIMARY
+              return true if ntfs.detect_content_info.windows?
+            end
+          end
+        end
```

Comment:
I fully agree with the fact that libstorage-ng is programmed in a very agnostic way. A filesystem can spread on several block devices and all those kind of (not so) crazy combinations. But I'm not sure if I like the idea of having so much "low level" code in yast2-country. Maybe yast2-storage-ng should provide some more high level methods. Otherwise, the internal structure of libstorage-ng (with things like all those casts that look so unnatural in Ruby) will spread all over the YaST code.